core/server/lib/members/static/preact/index.js

```diff
@@ -7,7 +7,7 @@ const storage = window.localStorage;
 var layer0 = require('./layer0');
 
 function getFreshState() {
-    const [hash, formType, query] = window.location.hash.match(/^#(?=\S*['-])([a-zA-Z'-]+)\??(.*)$/) || ['#signin?', 'signin', ''];
+    const [hash, formType, query] = window.location.hash.match(/^#(\w[\w']*(?:-\w+)*)\??(.*)$/) || ['#signin?', 'signin', ''];
```

Comment:
nabd but could this regex be smth like
```
/^#([^?]+)\??(.*($/
```

AKA match anything that isn't a `?` - it might be a little easier to grok